src/ProgressOnderwijsUtils/Win32/VirusScan.cs

```diff
@@ -0,0 +1,55 @@
+﻿#nullable disable annotations
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.Windows.Sdk;
+using ProgressOnderwijsUtils;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class VirusScan
+    {
+        public static bool IsMalware(string content, string contentName)
+        {
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanString(scanSession.Context, content, contentName, IntPtr.Zero, out var result);
+            return ResultIsMalware(result);
+        }
+
+        public static bool IsMalware(byte[] buffer, string contentName) => IsMalwareUnsafe(buffer, contentName);
+
+        static unsafe bool IsMalwareUnsafe(byte[] buffer, string contentName)
+        {
+            var sizet = Marshal.SizeOf(typeof(byte)) * buffer.Length;
+            var bufferPtr = Marshal.AllocHGlobal(sizet).ToPointer();
+
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanBuffer(scanSession.Context, bufferPtr, (uint)buffer.LongLength, contentName, IntPtr.Zero, out var result);
+            return ResultIsMalware(result);
+        }

```

Comment:
Ik denk dat dit duidelijk te tricky is. De oppervlakkige fix is om wel netjes alles op te ruimen en je nieuwe buffer wel te vullen; maar dat is onnodig tricky; de veel makkelijkere aanpak is om `fixed` te gebruiken - maar *eigenlijk* vraag ik me af of we hier niet gewoon beter de interop laag het marshallen moeten laten doen, om dit soort tricky bugs minder mogelijk te maken.

Oftewel, de oude aanpak.

Reply:
dus virusscan niet verhuizen naar utils? Of de oude virusscan.cs naar utils overzetten

Reply:
naja, misschien is de `fixed` oplossing wel een prima tussenvariant; de win ten op zichte van de versie in progress is dan dat we niet zelf de DllImport's hoeven maken/bijhouden.

Mij om het even, maar kan prima naar utils nu je toch bezig bent.

Reply:
Dit is de fixed die ik bedoel, trouwens: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/fixed-statement

Reply:
ik had nog nooit van het fixed statement gehoord. maar zie er nu wat voorbeelden van. Ik zal kijken of ik er wat van kan maken.

Reply:
ja die had ik idd net gevonden ;)

Reply:
Het is iig aanzienlijk minder gedoe dan dat hele AllocHGlobal gedoe ;-)

Reply:
en de test werkt nu ook. met de 'fixed'